plugins/inputs/kinesis_consumer/kinesis_consumer.go

```diff
@@ -237,7 +251,42 @@ func (k *KinesisConsumer) Start(ac telegraf.Accumulator) error {
 	return nil
 }
 
+func (k *KinesisConsumer) decompress(input []byte) ([]byte, error) {
+	var decompressedData []byte
+	var err error
+
+	switch k.CompressionType {
+	case compressionSnappy:
+		decompressedData, err = decompressSnappy(input)
+		if err != nil {
+			return nil, err
+		}
+	case compressionGZIP:
+		decompressedData, err = decompressGZip(input)
+		if err != nil {
+			return nil, err
+		}
+	default:
```

Comment:
Default to uncompressed and set `decompressedData = input`; add a case for a compression type of "none" and do the same.

Reply:
I'm happy to do this. Can we do the same for the output plugin in that case. #5588